crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -61,6 +67,9 @@ impl<T: Clone + Resource> State<T> {
         Wrapper::<T, OnExit>::new(discriminant(&d))
     }
 
+    /// Creates a driver set for the State.
+    ///
+    /// Important note: this set must be inserted **before** all other state-dependant sets to work properly!
```

Comment:
This is unfortunate, but I don't think this can be avoided until #1375 or some stop-gap measure that lets us order run criteria evaluation.